craft_parts/state_manager/step_state.py

```diff
@@ -0,0 +1,121 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2016-2021 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""The step state preserves step execution context information."""
+
+import os
+from abc import ABC, abstractmethod
+from pathlib import Path
+from typing import Any, Dict, Set
+
+from pydantic import Field
+from pydantic_yaml import YamlModel  # type: ignore
+
+
+class StepState(YamlModel, ABC):
+    """The context used to run a step for a part.
```

Comment:
"is used"?

Reply:
Rephrased that, we don't actually "use" a context to run a step.